robyn/__init__.py

```diff
@@ -165,6 +166,40 @@ def terminating_signal_handler(_sig, _frame):
                 observer.stop()
                 observer.join()
 
+    def add_view(self, endpoint: str, view: Callable, const: bool = False):
+        """
+        [This is base handler for the view decorators]
+
+        :param endpoint [str]: [endpoint for the route added]
+        :param handler [function]: [represents the function passed as a parent handler for single route with different route types]
+        """
+        http_methods = {"GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS"}
```

Comment:
Make this a constant defined outside of the method (with the `Final` typing), like this:
```
HTTP_METHODS: Final[set[str]] = ...
```
This will allocate the set only once.

Reply:
@AntoineRR , unfortunately the `final` keyword was introduced in Python 3.8(https://peps.python.org/pep-0591/) and we still support 3.7 and above.

Also, the reallocation is only done during the add time, which will affect the startup but will have no performance impact on serving responses. So, I don't think we need a global variable just for this.


What do you reckon?

Reply:
Ok, we can leave it like this, this is not very important :)